File: Cplusplus/LEVEL 2/기능개발.cpp
//22.01.26 기능개발 by CJHofficial
#include <string>
#include <vector>
#include <cmath>
#include <map>

using namespace std;

vector<int> solution(vector<int> progresses, vector<int> speeds) {
    vector<int> answer;
    vector<int> needDayVec;
    int remainWork, needDay;
    int finishedWork = 1;
    map<int, int> result;
    for (int i = 0; i < progresses.size(); i++) {
        remainWork = 100 - progresses[i];
        needDay = ceil((double)remainWork / speeds[i]);
        needDayVec.push_back(needDay);
    }
    int ndvSize = needDayVec.size();
    for (int i = (ndvSize - 1); i >= 0; i--) {
        int MAX = needDayVec[i];
        for (int j = i - 1; j >= 0; j--) {
            if (MAX < needDayVec[j]) MAX = needDayVec[j];
        }
        result[MAX]++;
    }
    for (auto p : result) {
        answer.push_back(p.second);
    }
    return answer;
}

File: Python/level2/기능개발.py
def solution(progresses, speeds):
    answer = []
    stack = []
    for index, progress in enumerate(progresses):
        days = 0
        while progress + days * speeds[index] < 100:
            days += 1
        if len(stack) == 0:
            stack.append(days)
        else:
            if stack[0] >= days:
                stack.append(days)
            else:
                answer.append(len(stack))
                stack = [days]
    if len(stack):
        answer.append(len(stack))
    return answer
